include/rdma/fi_domain.h

```diff
@@ -369,7 +370,8 @@ static inline int
 fi_mr_refresh(struct fid_mr *mr, const struct iovec *iov, size_t count,
 	      uint64_t flags)
 {
-	struct fi_mr_modify modify = { 0 };
+	struct fi_mr_modify modify;
+	memset(&modify, 0, sizeof(modify));
```

Comment:
@shefty Does this function really need to be inlined?

I'm guessing this PR is because C++ doesn't support `struct foo bar = {0}`.  But if this function isn't inlined, that issue goes away (because C++ applications that include `fi_domain.h` won't see that code, and therefore will compile just fine).

Reply:
That is true, if the function does not need to be inlined, it does not need to be modified.  Any thoughts on whether it needs to be inlined?

Reply:
Technically, probably not.  But the function is defined similar to most other functions accessed by the apps, nearly all of which are just wrappers around accessing function pointers.  I would not want to make a special exception for this and add a new library export.